internal/service/searchindex/resource_search_index.go

```diff
@@ -396,38 +391,39 @@ func flattenSearchIndexSynonyms(synonyms []admin.SearchSynonymMappingDefinition)
 
 func marshalSearchIndex(fields any) (string, error) {
 	bytes, err := json.Marshal(fields)
-	return string(bytes), err
+	str := string(bytes)
+	if str == "[]" || str == "{}" { // empty JSON array or object
+		str = ""
+	}
+	return str, err
 }
 
-func resourceMongoDBAtlasSearchIndexCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	projectID := d.Get("project_id").(string)
 	clusterName := d.Get("cluster_name").(string)
 	indexType := d.Get("type").(string)
-
 	searchIndexRequest := &admin.ClusterSearchIndex{
 		Type:           conversion.StringPtr(indexType),
 		Analyzer:       conversion.StringPtr(d.Get("analyzer").(string)),
 		CollectionName: d.Get("collection_name").(string),
 		Database:       d.Get("database").(string),
 		Name:           d.Get("name").(string),
 		SearchAnalyzer: conversion.StringPtr(d.Get("search_analyzer").(string)),
-		Status:         conversion.StringPtr(d.Get("status").(string)),
```

Comment:
If status is read only we can remove the `Optional` property in the schema and leave it as `Computed`. Also saw it is currently not defined in the docs.

Reply:
I think the intention was not to expose it in the TF schema as not in the doc, is it ok if i remove it from the schema as it's not documented?